build_tools/travis/install.sh

```diff
@@ -25,12 +25,18 @@ if [[ ! -d /home/travis/miniconda3 ]]
     export PATH=/home/travis/miniconda3/bin:$PATH
     echo $PATH
     conda update --yes conda
-    conda create -n testenv --yes python=3.5
+    # If we are running pylint, use Python 3.5.2 due to
```

Comment:
Because of the `if [[ ! -d /home/travis/miniconda3 ]]` check at the top, this will only get run once, and so only one of these environments will ever get created, and then the `activate` line will fail when the cache gets loaded.  Does that make sense?  So I think you just need to have two `conda create -n ...` commands, like you do above, without any `ifs`.